src/zenml/model_registries/base_model_registry.py

```diff
@@ -0,0 +1,425 @@
+        registry_metadata: The metadata associated with this model version
+    """
+
+    model_registration: ModelRegistration
+    model_source_uri: str
+    version_description: Optional[str] = None
+    version: Optional[str] = None
+    created_at: Optional[int] = None
+    last_updated_at: Optional[int] = None
+    version_stage: ModelVersionStage = ModelVersionStage.NONE
+    version_tags: Dict[str, str] = Field(default_factory=dict)
+    registry_metadata: Dict[str, str] = Field(default_factory=dict)
+    zenml_metadata: Optional[ZenMLModelMetadata] = Field(
+        default_factory=ZenMLModelMetadata
+    )
+
+    @root_validator
+    def fill_in_out_zenml_metadata(
+        cls, values: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        """Fills in ZenML metadata if not provided.
+
+        Args:
+            values: The values to validate.
+
+        Returns:
+            The validated values.
+        """
+        if values.get("zenml_metadata") is None:
+            values["zenml_metadata"] = ZenMLModelMetadata()
+        version_tags = values.get("version_tags")
+        if version_tags:
+            if "zenml_version" in version_tags.keys():
+                values["zenml_metadata"].zenml_version = version_tags[
+                    "zenml_version"
+                ]
+                values["version_tags"].pop("zenml_version")
+            if "zenml_pipeline_run_id" in version_tags.keys():
+                values["zenml_metadata"].zenml_pipeline_run_id = version_tags[
+                    "zenml_pipeline_run_id"
+                ]
+                values["version_tags"].pop("zenml_pipeline_run_id")
+            if "zenml_pipeline_name" in version_tags.keys():
+                values["zenml_metadata"].zenml_pipeline_name = version_tags[
+                    "zenml_pipeline_name"
+                ]
+                values["version_tags"].pop("zenml_pipeline_name")
+            if "zenml_step_name" in version_tags.keys():
+                values["zenml_metadata"].zenml_step_name = version_tags[
+                    "zenml_step_name"
+                ]
+                values["version_tags"].pop("zenml_step_name")
+        return values
+
+
+class BaseModelRegistryConfig(StackComponentConfig):
+    """Base config for model registries."""
+
+
+class BaseModelRegistry(StackComponent, ABC):
+    """Base class for all ZenML model registries."""
+
+    @property
+    def config(self) -> BaseModelRegistryConfig:
+        """Returns the config of the model registries.
+
+        Returns:
+            The config of the model registries.
+        """
+        return cast(BaseModelRegistryConfig, self._config)
+
+    # ---------
+    # Model Registration Methods
+    # ---------
+
+    @abstractmethod
+    def register_model(
+        self,
+        name: str,
+        description: Optional[str] = None,
+        tags: Optional[Dict[str, str]] = None,
+    ) -> ModelRegistration:
+        """Registers a model in the model registry.
+
```

Comment:
You should set out some rules and expectations in the abstraction concerning what happens if you already have a model with that same name in the registry. Do you expect a KeyError to be raised ?